proximal/examples/test_deconv_sinv_psf.py

```diff
@@ -0,0 +1,89 @@
+# Override the default ProxImaL installation in the system
+import sys
+sys.path.append('../../')
+
+
+import proximal as px
+from proximal.lin_ops.black_box import LinOpFactory
+from scipy.signal import convolve
+
+#===========
+# SYNTHETIC DATA
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.stats import multivariate_normal
+
+# Create Synthetic Horizontal Stripes Pattern Image
+def get_raw(width=3840, height=2748):
+    im_s = np.zeros((2748, 3840))
+    stripe_width, stripe_gap, stripe_start, stripe_end = 50, 50, 500, 500
+    for y in range(0, im_s.shape[0]-(stripe_start+stripe_end), stripe_width + stripe_gap):
+        im_s[stripe_start+y : stripe_start+y + 50, :] = .8
+
+    xx, yy = np.mgrid[0:im_s.shape[0], 0:im_s.shape[1]]
+    im_ctr = (np.array(im_s.shape)/2).astype(int)
+    r = np.sqrt((xx - im_ctr[0])**2 + (yy - im_ctr[1])**2)
+    mask = np.zeros_like(im_s)
+    mask[r < 1000] = 1
+    im_s *= mask
+
+    return np.asfortranarray(im_s, dtype=np.float32), im_ctr
+
+im_s, im_ctr = get_raw()
+plt.subplot(221); plt.imshow(im_s); plt.title('Ground Truth');
+
+# Create Gaussian Kernel & Convolve
+def get_blur_kernel(im_shape, im_ctr, crop=True):
+    x, y = np.mgrid[0:im_shape[0], 0:im_shape[1]]
+    pos = np.dstack((x, y))
+    rv = multivariate_normal.pdf(pos, im_ctr, [[200, 0], [300, 500]])
+    psf2 = np.asfortranarray(rv/np.max(rv), dtype=np.float32)
+
+    if not crop:
+        return psf2
+
+    psf2 = psf2[im_ctr[0]-100:im_ctr[0]+101, im_ctr[1]-100:im_ctr[1]+101]
+
+    return psf2
+
+psf2 = get_blur_kernel(im_s.shape, im_ctr)
+plt.subplot(222); plt.imshow(psf2); plt.title('PSF zoomed in');
+
+im_sb = np.asfortranarray(convolve(im_s, psf2, mode='same', method='auto'), dtype=np.float32)
+plt.subplot(223); plt.imshow(im_sb); plt.title('Blurred Image'); plt.colorbar()
```

Comment:
> Also regarding the space-invariant case, I generated results using with px.conv() and then using BlackBox based convolution forward operator by toggling the using_px_conv flag, and the results are visibly bad with the latter. Do you know why?

I have a hypothesis, but I need to see the code to verify it. `px.conv()` assumes the 2D kernel and the sliding window is aligned at the top-left corner, i.e. `convolved(x, y) += kernel(dx, dy) * input(x + dx, y + dy) where dx, dy >= 0`. In contrast, `scipy...convolve` assumes the kernel is centered at the target pixel.

>  generated results using with px.conv()

Interesting... it is supposed to be a hidden feature / private function. May I know the purpose of using `px.conv()` to simulate low-resolution images?

Reply:
Another hypothesis: the convergence rate is lower when `using_px_conv=False`. Try commenting out `max_iters=20` and try again.

Reply:
I just did that and it ran for 140 iterations, which took like >30min, but still the result has some edge artifacts as shown:
<img width="529" alt="image" src="https://github.com/comp-imaging/ProxImaL/assets/4764586/fab60b4c-1ee0-4261-a010-71c4ed28d469">

Reply:
@shnaqvi Discussion moved to https://github.com/comp-imaging/ProxImaL/pull/84#discussion_r1323355846 .